src/DataFramesMeta.jl

```diff
@@ -103,74 +103,134 @@ macro col(kw)
     esc(fun_to_vec(kw))
 end
 
+is_simple_function_call(x) = false
+function is_simple_function_call(expr::Expr)
+    (expr.head == :call
+        && length(expr.args) >= 2
+        && expr.args[1] isa Symbol
+        && all(x -> x isa QuoteNode, expr.args[2:end]))
+end
+
+is_simple_broadcast_call(x) = false
+function is_simple_broadcast_call(expr::Expr)
+    (expr.head == :.
+        && length(expr.args) == 2
+        && expr.args[1] isa Symbol
+        && expr.args[2] isa Expr
+        && expr.args[2].head == :tuple
+        && all(x -> x isa QuoteNode, expr.args[2].args))
+end
+
+
+function broadcast_f_expr(f)
+    # theoretically we could use ByRow everywhere, but maybe we can save some compilation
+    # here and there by using the Base functionality where it's available, as other code
+    # might have already called that
+    if VERSION >= v"1.6.0-beta"
+        :(Base.BroadcastFunction($f))
+    else
+        :(DataFrames.ByRow($f))
+    end
+end
```

Comment:
@bkamins Do you think there would be any advantage to using `ByRow`? Maybe we will automatically use multithreading with `ByRow` in the future?

Reply:
`ByRow` is and I think should be just a wrapper for broadcasting. If broadcasting uses multithreading in the future it will automatically make `ByRow` use it.

Though I think that compilation time savings are minimal, as `ByRow` is very cheap to compile.

Reply:
It's all `ByRow` now.